File: src/vanilla/DanModel/JavaClass.cs
using System;

namespace AutoRest.Java.DanModel
{
    public class JavaClass : JavaAbstractType
    {
        private readonly JavaFileContents contents;
        private bool addNewLine;

        public JavaClass(JavaFileContents contents)
        {
            this.contents = contents;
        }

        private void AddExpectedNewLine()
        {
            if (addNewLine)
            {
                contents.Line();
                addNewLine = false;
            }
        }

        public void PrivateMemberVariable(string variableType, string variableName)
        {
            PrivateMemberVariable($"{variableType} {variableName}");
        }

        public void PrivateMemberVariable(string variableDeclaration)
        {
            AddExpectedNewLine();
            contents.Line($"private {variableDeclaration};");
            addNewLine = true;
        }

        public void PrivateFinalMemberVariable(string variableType, string variableName)
        {
            AddExpectedNewLine();
            contents.Line($"private final {variableType} {variableName};");
            addNewLine = true;
        }

        public void PublicStaticFinalVariable(string variableDeclaration)
        {
            AddExpectedNewLine();
            contents.Line($"public static final {variableDeclaration};");
            addNewLine = true;
        }

        public void PublicConstructor(string constructorSignature, Action<JavaBlock> constructor)
        {
            AddExpectedNewLine();
            contents.Block($"public {constructorSignature}", constructor);
            addNewLine = true;
        }

        public void PublicMethod(string methodSignature, Action<JavaBlock> method)
        {
            AddExpectedNewLine();
            contents.Block($"public {methodSignature}", method);
            addNewLine = true;
        }

        public void ProtectedMethod(string methodSignature, Action<JavaBlock> method)
        {
            AddExpectedNewLine();
            contents.Block($"protected {methodSignature}", method);
            addNewLine = true;
        }

        public void PrivateMethod(string methodSignature, Action<JavaBlock> method)
        {
            AddExpectedNewLine();
            contents.Block($"private {methodSignature}", method);
            addNewLine = true;
        }

        public void PublicStaticMethod(string methodSignature, Action<JavaBlock> method)
        {
            AddExpectedNewLine();
            contents.Block($"public static {methodSignature}", method);
            addNewLine = true;
        }

        public void PublicInterface(string interfaceSignature, Action<JavaInterface> interfaceBlock)
        {
            AddExpectedNewLine();
            contents.PublicInterface(interfaceSignature, interfaceBlock);
            addNewLine = true;
        }

        public void Interface(string interfaceSignature, Action<JavaInterface> interfaceBlock)
        {
            AddExpectedNewLine();
            contents.Interface(interfaceSignature, interfaceBlock);
            addNewLine = true;
        }

        public void PrivateStaticFinalClass(string classSignature, Action<JavaClass> classBlock)
        {
            AddExpectedNewLine();
            contents.PrivateStaticFinalClass(classSignature, classBlock);
            addNewLine = true;
        }

        public void MultipleLineComment(string description)
        {
            MultipleLineComment(comment =>
            {
                comment.Description(description);
            });
        }

        public void MultipleLineComment(Action<JavaMultipleLineComment> commentAction)
        {
            AddExpectedNewLine();
            contents.MultipleLineComment(commentAction);
        }

        public void WordWrappedMultipleLineComment(int wordWrapWidth, Action<JavaMultipleLineComment> commentAction)
        {
            AddExpectedNewLine();
            contents.WordWrappedMultipleLineComment(wordWrapWidth, commentAction);
        }

        public void Annotation(params string[] annotations)
        {
            AddExpectedNewLine();
            contents.Annotation(annotations);
        }
    }
}

File: mapper/src/main/java/com/azure/autorest/model/javamodel/JavaClass.java
package com.azure.autorest.model.javamodel;

public class JavaClass implements JavaType
{
    private JavaFileContents contents;
    private boolean addNewLine;

    public JavaClass(JavaFileContents contents)
    {
        this.contents = contents;
    }

    private void AddExpectedNewLine()
    {
        if (addNewLine)
        {
            contents.Line();
            addNewLine = false;
        }
    }

    public final void PrivateMemberVariable(String variableType, String variableName)
    {
        PrivateMemberVariable(String.format("%1$s %2$s", variableType, variableName));
    }

    public final void PrivateMemberVariable(String variableDeclaration)
    {
        AddExpectedNewLine();
        contents.Line(String.format("private %1$s;", variableDeclaration));
        addNewLine = true;
    }

    public final void PrivateFinalMemberVariable(String variableType, String variableName)
    {
        AddExpectedNewLine();
        contents.Line(String.format("private final %1$s %2$s;", variableType, variableName));
        addNewLine = true;
    }

    public final void PublicStaticFinalVariable(String variableDeclaration)
    {
        AddExpectedNewLine();
        contents.Line(String.format("public static final %1$s;", variableDeclaration));
        addNewLine = true;
    }

    public final void PrivateConstructor(String constructorSignature, Consumer<JavaBlock> constructor)
    {
        AddExpectedNewLine();
        contents.Block(String.format("private %1$s", constructorSignature), constructor);
        addNewLine = true;
    }

    public final void PublicConstructor(String constructorSignature, Consumer<JavaBlock> constructor)
    {
        AddExpectedNewLine();
        contents.Block(String.format("public %1$s", constructorSignature), constructor);
        addNewLine = true;
    }

    public final void PackagePrivateConstructor(String constructorSignature, Consumer<JavaBlock> constructor)
    {
        AddExpectedNewLine();
        contents.Block(String.format("%1$s", constructorSignature), constructor);
        addNewLine = true;
    }

    public final void Method(JavaVisibility visibility, List<JavaModifier> modifiers, String methodSignature, Consumer<JavaBlock> method)
    {
        AddExpectedNewLine();
        contents.Method(visibility, modifiers, methodSignature, method);
        addNewLine = true;
    }

    public final void PublicMethod(String methodSignature, Consumer<JavaBlock> method)
    {
        Method(JavaVisibility.Public, null, methodSignature, method);
    }

    public final void PackagePrivateMethod(String methodSignature, Consumer<JavaBlock> method)
    {
        Method(JavaVisibility.PackagePrivate, null, methodSignature, method);
    }

    public final void PrivateMethod(String methodSignature, Consumer<JavaBlock> method)
    {
        Method(JavaVisibility.Private, null, methodSignature, method);
    }

    public final void PublicStaticMethod(String methodSignature, Consumer<JavaBlock> method)
    {
        Method(JavaVisibility.Public, new AutoRest.Java.JavaModifier[] {JavaModifier.Static}, methodSignature, method);
    }

    public final void Interface(JavaVisibility visibility, String interfaceSignature, Consumer<JavaInterface> interfaceBlock)
    {
        AddExpectedNewLine();
        contents.Interface(visibility, interfaceSignature, interfaceBlock);
        addNewLine = true;
    }

    public final void PublicInterface(String interfaceSignature, Consumer<JavaInterface> interfaceBlock)
    {
        Interface(JavaVisibility.Public, interfaceSignature, interfaceBlock);
    }

    public final void PrivateStaticFinalClass(String classSignature, Consumer<JavaClass> classBlock)
    {
        AddExpectedNewLine();
        contents.Class(JavaVisibility.Private, new AutoRest.Java.JavaModifier[] {JavaModifier.Static, JavaModifier.Final}, classSignature, classBlock);
        addNewLine = true;
    }

    public final void BlockComment(String description)
    {
        AddExpectedNewLine();
        contents.BlockComment(description);
    }

    public final void BlockComment(Consumer<JavaLineComment> commentAction)
    {
        AddExpectedNewLine();
        contents.BlockComment(commentAction);
    }

    public final void BlockComment(int wordWrapWidth, Consumer<JavaLineComment> commentAction)
    {
        AddExpectedNewLine();
        contents.BlockComment(wordWrapWidth, commentAction);
    }

    public final void JavadocComment(String description)
    {
        AddExpectedNewLine();
        contents.JavadocComment(description);
    }

    public final void JavadocComment(Consumer<JavaJavadocComment> commentAction)
    {
        AddExpectedNewLine();
        contents.JavadocComment(commentAction);
    }

    public final void JavadocComment(int wordWrapWidth, Consumer<JavaJavadocComment> commentAction)
    {
        AddExpectedNewLine();
        contents.JavadocComment(wordWrapWidth, commentAction);
    }

    public final void Annotation(String... annotations)
    {
        AddExpectedNewLine();
        contents.Annotation(annotations);
    }
}